packages/shared/src/utils/os/os_detect.ts

```diff
@@ -82,3 +82,19 @@ export const mobileOSDetect = () => {
 
     return 'unknown';
 };
+
+export const checkUserBrowser = () => {
+    // We can't rely only on navigator.userAgent.index, the verification order is also important
+    if ((navigator.userAgent.indexOf('Opera') || navigator.userAgent.indexOf('OPR')) !== -1) {
+        return 'Opera';
+    } else if (navigator.userAgent.indexOf('Edg') !== -1) {
+        return 'Edge';
+    } else if (navigator.userAgent.indexOf('Chrome') !== -1) {
+        return 'Chrome';
+    } else if (navigator.userAgent.indexOf('Safari') !== -1) {
+        return 'Safari';
+    } else if (navigator.userAgent.indexOf('Firefox') !== -1) {
+        return 'Firefox';
```

Comment:
it's better to not export it directly but add an object with methods that would incapsulate all these Firefox/Safari strings, and export it instead:
```
export const user_browser = {
    // We can't rely only on navigator.userAgent.index, the verification order is also important
    isOpera: () => checkUserBrowser() === 'Opera',
    isEdge: () => checkUserBrowser() === 'Edge',
    isChrome: () => checkUserBrowser() === 'Chrome',
    isSafari: () => checkUserBrowser() === 'Safari',
    isFirefox: () => checkUserBrowser() === 'Firefox',
```
and use it in your component like this: `user_browser.isSafari();`